# HG changeset patch
# User Marek Marczykowski <marmarek@mimuw.edu.pl>
# Date 1307662302 -7200
# Node ID fd3be19373206355fab6e2dd36ea0e14097f5873
# Parent  9d58dbe3e848401883b3b6720b8f28323f10ac6f
libxl: do not try to detach pci device when none attached

Fixes SEGV on domain destroy

Signed-off-by: Marek Marczykowski <marmarek@mimuw.edu.pl>

diff -r 9d58dbe3e848 -r fd3be1937320 tools/libxl/libxl_pci.c
--- a/tools/libxl/libxl_pci.c	Fri Jun 10 01:11:01 2011 +0200
+++ b/tools/libxl/libxl_pci.c	Fri Jun 10 01:31:42 2011 +0200
@@ -1075,6 +1075,8 @@ int libxl_device_pci_shutdown(libxl_ctx 
     rc = libxl_device_pci_list_assigned(ctx, &pcidevs, domid, &num);
     if ( rc )
         return rc;
+    if ( num == 0 )
+        return 0;
     for (i = 0; i < num; i++) {
         /* Force remove on shutdown since, on HVM, qemu will not always
          * respond to SCI interrupt because the guest kernel has shut down the
exporting patch:
<fdopen>
